### DISCUSSION TITLE
Queue messages for slow mode

### DISCUSSION THREAD

**Discussion by erinclemmer** [Chatterino/chatterino2#4079] [Ideas]
I saw [this issue](https://github.com/night/betterttv/issues/5629) on the better ttv repo and it seemed pretty cool. However, I don't think it's possible using a chrome extension because it would require relying on elements that aren't guaranteed to be there or could have classes/ids that change over time (slow icon and countdown) and it would have to stop the execution of sending messages and pass it through a new function. I think this could be implemented in chatterino without too much hassle. I don't mind tackling it as a project but I want to gauge the interest or feasibility on here first.

**Comment by pajlada** [comment]
I'm not 100% sure of how nice of a user-experience this would be, but some of the steps required to implement this would definitely be nice to have. Namely information about when you can send your next message. This could be used to indicate that you can't type for another X seconds and will be an improvement is usability from what we have now.
With some of the more basic components implemented, the functionality shouldn't be super complicated so it'd be worth prototyping

**Reply by erinclemmer** [reply]
@pajlada Yeah, that's a good idea. At least having the countdown would be in parity with the web app.

**Reply by Mm2PL** [reply]
> This could be used to indicate that you can't type for another X seconds and will be an improvement is usability from what we have now.

I remember there being a patch for a countdown but it had problems (untimeout being the main one), however I think there is a way to make this happen. Perhaps the `new messages below` widget could be reused for this functionality.

I would personally like messages to be buffered slightly if it's impossible to send them right now but the next moment it will be possible.

**Reply by LosFarmosCTL** [reply]
#3329 is the mentioned PR for the timer, but it has been stale for over a year now.

**Reply by erinclemmer** [reply]
@LosFarmosCTL That's cool, I could redo the PR so that it gets updated from master and ready to be merged. Not sure what needs to be done with it other than some theme tweaks. I think @pajlada was going to get back with a code review but it never happened.
